src/EnergyPlus/GroundHeatExchangers.cc

```
@@ -492,27 +486,14 @@ GLHEResponseFactors::GLHEResponseFactors(EnergyPlusData &state, std::string cons
         tmpGvals.push_back(var.at("g_function_g_value").get<Real64>());
     }
 
-    bool errorsFound = false;
-
-    if (tmpLntts.size() == tmpLntts.size()) {
```

OK, so this is an odd one.  The GLHE needs an input of matched pairs of t and g.  What happens if the user enters a missing trailing G?  

I tested it, and as expected nowadays, the mismatched extensible pair is caught by the input processor and it fails early and gracefully.  This check was originally intended to catch the error gracefully and is no longer needed.  And of course, it didn't do anything as it was anyway, it was just an accidental typo when the routine was converted to read JSON inputs: https://github.com/NREL/EnergyPlus/commit/d4b8582aac7f0a039911549ca6f1f33f6a7e052b

---

It's good to see where that check came from. After looking at this and seeing the extensible attribute I do wonder if there is an input configuration that could cause an issue.

---

Maybe, but I pulled the test file, cut off the trailing G, and it failed nicely.  I think it's protected sufficiently for now.